includes/discount-functions.php

```diff
@@ -905,11 +905,12 @@ function edd_is_discount_valid( $code = '', $user = '', $set_error = true ) {
  * @since 3.0 Updated to call edd_get_discount_by_code()
  *
  * @param string $code Discount code.
- * @return int Discount ID.
+ * @return int|bool Discount ID, or false if discount does not exist.
  */
 function edd_get_discount_id_by_code( $code = '' ) {
 	$discount = edd_get_discount_by_code( $code );
-	return $discount->id;
+
+	return is_object( $discount ) ? $discount->id : false;
```

Comment:
@robincornett I think checking for a `false` value or an `instanceof EDD_Discount` seems more appropriate since those are the two expected return values of `edd_get_discount_by_code()`

Reply:
Thanks, the the `instanceof` check is one I've only used once before and I couldn't remember it. Updated.